py4syn/epics/OceanClass.py

```diff
@@ -100,6 +96,14 @@ def setCountStop(self):
 
     def saveSpectrum(self, **kwargs):
         ''' save the spectrum intensity in a mca file or an hdf file '''
+        dark = self.pvDarkCorrection.get()
+
+        # the spectra come from different pv if use darkcorrection
+        if dark == 1:
+            self.pvSpectrum = PV(self.pv+":DarkCorrectedSpectra")
```

Comment:
Please instead of creating the PV in here, do it at __init__ and do get on the proper object.
This will mean that every time that you try to save a Spectrum it will try to create a new connection and check PyEpics cache and etc.

Change to something like this:
```python
if self.pvDarkCorrection.get() == 1:
    self.spectrum = self.pvDarkCorrectedSpectra.get(count=self.numPoints, as_numpy=True)
else:
    self.spectrum = self.pvSpectra.get(count=self.numPoints, as_numpy=True)
```